interactive_engine/compiler/src/main/java/com/alibaba/graphscope/gremlin/integration/suite/standard/IrGremlinQueryTest.java

```diff
@@ -428,6 +434,57 @@ public void g_V_haslabel_union_identity_out_values() {
         Assert.assertEquals(4, counter);
     }
 
+    @LoadGraphWith(LoadGraphWith.GraphData.MODERN)
+    @Test
+    public void g_V_fold_unfold_values() {
+        Traversal<Vertex, Object> traversal = this.get_g_V_fold_unfold_values();
+        this.printTraversalForm(traversal);
+        int counter = 0;
+
+        List<String> expected = Arrays.asList("1", "2", "3", "4", "5", "6");
+
+        while (traversal.hasNext()) {
+            Object result = traversal.next();
+            Assert.assertTrue(expected.contains(result.toString()));
+            ++counter;
+        }
+        Assert.assertEquals(6, counter);
+    }
+
+    @LoadGraphWith(LoadGraphWith.GraphData.MODERN)
+    @Test
+    public void g_V_fold_a_unfold_values() {
+        Traversal<Vertex, Object> traversal = this.get_g_V_fold_a_unfold_values();
```

Comment:
Can we check whether the tag `a` is still in the record of the unfolded results?